src/Microsoft.AspNetCore.ResponseCompression/DefaultMimeTypeFilterConfig.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace Microsoft.AspNetCore.ResponseCompression
+{
+    /// <inheritdoc/>
+    public class DefaultMimeTypeFilterConfig : IMimeTypeFilterConfig
+    {
+        private readonly HashSet<string> _compressed = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+        private readonly HashSet<string> _notCompressed = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+        /// <inheritdoc/>
+        public IEnumerable<string> CompressedMimeTypes => _compressed;
```

Comment:
If these sets were exposed as mutable collections then all of the methods could be removed from this type and become extension methods. They're all syntax sugar over the collections and I'm not sure we need most of them. The primary operation will be to add one or more entries to either list.

Reply:
This collection is practically the same as Options.MimeTypes. We could consolidate the storage to the base Options class with two properties, the existing MimeTypes and a new ExcludedMimeTypes. Then IMimeTypeFilterConfig goes away entirely.